mesos_master/datadog_checks/mesos_master/mesos_master.py

```diff
@@ -158,33 +158,39 @@ def __init__(self, name, init_config, instances):
 
     def _get_json(self, url, failure_expected=False, tags=None):
         tags = tags + ["url:%s" % url] if tags else ["url:%s" % url]
+
+        response, msg, status = self._make_request(url)
+
+        self.log.debug('Request to url : %s, timeout: %s, message: %s', url, self.http.options['timeout'], msg)
+        self._send_service_check(url, status, failure_expected=failure_expected, tags=tags, message=msg)
+
+        if response.encoding is None:
+            response.encoding = 'UTF8'
+
+        return response.json()
+
+    def _make_request(self, url):
         msg = None
         status = None
-        timeout = self.http.options['timeout']
         response = None
+
         try:
             response = self.http.get(url)
             if response.status_code != 200:
                 status = AgentCheck.CRITICAL
-                msg = "Got %s when hitting %s" % (response.status_code, url)
+                msg = "Got {} when hitting {}".format(response.status_code, url)
             else:
```

Comment:
we wouldn't get a 307, because we make the HTTP requests with `allow_redirects=True`
That's why we check the request history to find a possible 307 https://github.com/DataDog/integrations-core/pull/9500/files#diff-b3313b226583032aeb26e68e9b095e9cfcd93cc45727712529fa56c8313c4e91R210

when `_get_master_state` returns None, `self.leader` is set to false and the rest of the check remains the same